runtime-transaction/src/builtin_auxiliary_data_store.rs

```diff
@@ -0,0 +1,104 @@
+// static account keys has max
+use {
+    agave_transaction_view::static_account_keys_meta::MAX_STATIC_ACCOUNTS_PER_PACKET as FILTER_SIZE,
+    solana_builtins_default_costs::{BUILTIN_INSTRUCTION_COSTS, MAYBE_BUILTIN_KEY},
+    solana_sdk::pubkey::Pubkey,
+};
+
+pub(crate) struct BuiltinAuxiliaryDataStore {
+    // Array of auxiliary data for all possible static and sanitized program_id_index,
+    // Each possible value of data indicates:
+    //   None - un-checked
+    //   Some<None> - checked, not builtin
+    //   Some<Some<(bool, u32)>> - checked, is builtin and (is-compute-budget, default-cost)
```

Comment:
It seems to me this would better represented by a new enum:

```rust
    #[derive(Default)]
    enum BuiltinCheckStatus {
		#[default]
        Unchecked,
        NotBuiltin,
        Builtin{
            is_compute_budget: bool,
            default_cost: u32,
        }
    }
```

Reply:
think this saves a byte as well since we don't need 2 option discriminants

Reply:
enum is a way to go

Reply:
d76291d6af7170e630bb6a3357de9b11231f217b

<details>
<summary>savings cross all benches, due to smaller memory footprint</summary>

```
     Running benches/process_compute_budget_instructions.rs (target/release/deps/process_compute_budget_instructions-b53a7f97abe58117)
bench_process_compute_budget_instructions_empty/0 instructions
                        time:   [11.957 µs 11.975 µs 11.995 µs]
                        thrpt:  [85.368 Melem/s 85.513 Melem/s 85.641 Melem/s]
                 change:
                        time:   [-29.911% -29.703% -29.488%] (p = 0.00 < 0.05)
                        thrpt:  [+41.820% +42.253% +42.676%]
                        Performance has improved.
Found 7 outliers among 100 measurements (7.00%)
  5 (5.00%) high mild
  2 (2.00%) high severe

bench_process_compute_budget_instructions_no_builtins/4 dummy Instructions
                        time:   [18.924 µs 18.941 µs 18.959 µs]
                        thrpt:  [54.011 Melem/s 54.062 Melem/s 54.112 Melem/s]
                 change:
                        time:   [-15.416% -15.210% -15.005%] (p = 0.00 < 0.05)
                        thrpt:  [+17.654% +17.938% +18.225%]
                        Performance has improved.
Found 6 outliers among 100 measurements (6.00%)
  1 (1.00%) low mild
  1 (1.00%) high mild
  4 (4.00%) high severe

bench_process_compute_budget_instructions_compute_budgets/4 compute-budget instructions
                        time:   [37.252 µs 37.332 µs 37.424 µs]
                        thrpt:  [27.362 Melem/s 27.430 Melem/s 27.488 Melem/s]
                 change:
                        time:   [-21.462% -21.177% -20.898%] (p = 0.00 < 0.05)
                        thrpt:  [+26.419% +26.867% +27.328%]
                        Performance has improved.
Found 2 outliers among 100 measurements (2.00%)
  2 (2.00%) high mild

bench_process_compute_budget_instructions_builtins/4 dummy builtins
                        time:   [41.314 µs 41.521 µs 41.730 µs]
                        thrpt:  [24.539 Melem/s 24.662 Melem/s 24.786 Melem/s]
                 change:
                        time:   [-8.4225% -8.0835% -7.7755%] (p = 0.00 < 0.05)
                        thrpt:  [+8.4311% +8.7944% +9.1972%]
                        Performance has improved.
Found 18 outliers among 100 measurements (18.00%)
  18 (18.00%) high mild

bench_process_compute_budget_instructions_mixed/355 mixed instructions
                        time:   [539.98 µs 540.43 µs 540.90 µs]
                        thrpt:  [1.8931 Melem/s 1.8948 Melem/s 1.8964 Melem/s]
                 change:
                        time:   [-2.9610% -2.8023% -2.6445%] (p = 0.00 < 0.05)
                        thrpt:  [+2.7164% +2.8831% +3.0513%]
                        Performance has improved.
Found 3 outliers among 100 measurements (3.00%)
  1 (1.00%) high mild
  2 (2.00%) high severe

bench_process_compute_budget_and_transfer_only/355 transfer instructions and compute budget ixs
                        time:   [607.07 µs 607.59 µs 608.19 µs]
                        thrpt:  [1.6837 Melem/s 1.6853 Melem/s 1.6868 Melem/s]
                 change:
                        time:   [-4.9817% -4.8051% -4.6671%] (p = 0.00 < 0.05)
                        thrpt:  [+4.8956% +5.0477% +5.2429%]
                        Performance has improved.
Found 7 outliers among 100 measurements (7.00%)
  5 (5.00%) high mild
  2 (2.00%) high severe

```

</details>